src/os/bluestore/BlueFS.cc

```diff
@@ -3730,17 +3732,37 @@ int BlueFS::_allocate(uint8_t id, uint64_t len,
                       bool permit_dev_fallback)
 {
   dout(10) << __func__ << " len 0x" << std::hex << len
-           << " alloc unit hint 0x" << alloc_unit
-           << std::dec << " from " << (int)id << dendl;
+           << " au 0x" << alloc_unit
+           << std::dec << " from " << (int)id
+           << " cooldown " << cooldown_deadline
+           << dendl;
   ceph_assert(id < alloc.size());
   int64_t alloc_len = 0;
   PExtentVector extents;
   uint64_t hint = 0;
   int64_t need = len;
+  bool shared = is_shared_alloc(id);
+  auto shared_unit = shared_alloc ? shared_alloc->alloc_unit : 0;
+  bool was_cooldown = false;
   if (alloc[id]) {
     if (!alloc_unit) {
       alloc_unit = alloc_size[id];
     }
+    // do not attempt shared_allocator with bluefs alloc unit
```

Comment:
My opinion is that if we went so far as to allow for 4K allocations go into BlueFS then we should not have a intermittent step of 64K.
My _feeling_ is that if we first require 64K alignment, and then drop it to 4K, we will produce more fragmentation, then if we asked 4K alignment from the beginning. 
We just should always request large chunks (~1M) and let the allocator provide us largest possible extents.

The cooldown approach makes perfect sense if 
1) we expect that after era of disk nearfull and fragmentation there would be an era of large empty segments
and
2) we know that reading 64K aligned data is faster then 4K aligned data (provided that we read large chunks in both cases)

Reply:
>My opinion is that if we went so far as to allow for 4K allocations go into BlueFS then we should not have a intermittent step of 64K.
My feeling is that if we first require 64K alignment, and then drop it to 4K, we will produce more fragmentation, then if we asked 4K alignment from the beginning.
We just should always request large chunks (~1M) and let the allocator provide us largest possible extents.
I'm a bit hesitant to use 4K permanently as we still don't fully understand how it affects performance. 

So you're suggesting to remove bluefs_shared_alloc_size parameter completely and apply bluefs_alloc_size unconditionally by default and fallback to 4K when necessary?  Or go even further and use 4K always?

>The cooldown approach makes perfect sense if
My primary intention was to 
1) avoid  costly long chunk allocation attempts if they failed at some point.
2) have means to automatically undo 1)

So if we still keep 1M/4K chunk size options then we better have that cooldown stuff..

Reply:
Your change is more aligned with current implementation.
We will have BlueFS working as-is up until we are low on shared block device.

I do retract my concept here until I do tests if having full 4K for fallback to shared block device is profitable.